models/iaf_wang_2002_exact.h

```diff
@@ -0,0 +1,516 @@
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef IAF_WANG_2002_EXACT
+#define IAF_WANG_2002_EXACT
+
+// Generated includes:
+#include "config.h"
+
+#ifdef HAVE_GSL
+
+// C includes:
+#include <gsl/gsl_errno.h>
+#include <gsl/gsl_matrix.h>
+#include <gsl/gsl_odeiv.h>
+
+// Includes from nestkernel:
+#include "archiving_node.h"
+#include "connection.h"
+#include "event.h"
+#include "nest_types.h"
+#include "ring_buffer.h"
+#include "universal_data_logger.h"
+
+namespace nest
+{
+/**
+ * Function computing right-hand side of ODE for GSL solver.
+ * @note Must be declared here so we can befriend it in class.
+ * @note Must have C-linkage for passing to GSL. Internally, it is
+ *       a first-class C++ function, but cannot be a member function
+ *       because of the C-linkage.
+ * @note No point in declaring it inline, since it is called
+ *       through a function pointer.
+ * @param void* Pointer to model neuron instance.
+ **/
+extern "C" inline int iaf_wang_2002_exact_dynamics( double, const double y[], double f[], void* pnode );
+
+
+/* BeginUserDocs: neuron, integrate-and-fire, conductance-based
+
+Short description
++++++++++++++++++
+
+Leaky integrate-and-fire-neuron model with conductance based synapses, and additional NMDA receptors.
+
+Description
++++++++++++
+
+``iaf_wang_2002_exact`` is a leaky integrate-and-fire neuron model with
+
+* an exact implementation of the neuron model described in [1]_.
+* exponential conductance-based AMPA and GABA-synapses
+* NMDA synapses with slow nonlinear dynamics
+* a fixed refractory period
+* no adaptation mechanisms
+
+Neuron and synaptic dynamics
+..................................................
+
+The membrane potential and synaptic variables evolve according to
+
+.. math::
+
+    C_\mathrm{m} \frac{dV(t)}{dt} &= -g_\mathrm{L} (V(t) - V_\mathrm{L}) - I_\mathrm{syn} (t) \\[3ex]
+    I_\mathrm{syn}(t) &= I_\mathrm{AMPA}(t) + I_\mathrm{NMDA}(t) + I_\mathrm{GABA}(t) (t) \\[3ex]
+    I_\mathrm{AMPA} &= (V(t) - V_E)\sum_{j \in \Gamma_\mathrm{ex}}^{N_E}w_jS_{j,\mathrm{AMPA}}(t) \\[3ex]
+    I_\mathrm{NMDA} &= \frac{(V(t) - V_E)}{1+[\mathrm{Mg^{2+}}]\mathrm{exp}(-0.062V(t))/3.57}\sum_{j \in
+    \Gamma_\mathrm{ex}}^{N_E}w_jS_{j,\mathrm{NMDA}}(t) \\[3ex]
+    I_\mathrm{GABA} &= (V(t) - V_I)\sum_{j \in \Gamma_\mathrm{in}}^{N_E}w_jS_{j,\mathrm{GABA}}(t) \\[5ex]
+    \frac{dS_{j,\mathrm{AMPA}}}{dt} &=-\frac{j,S_{\mathrm{AMPA}}}{\tau_\mathrm{AMPA}}+\sum_{k \in \Delta_j} \delta (t -
+t_j^k) \\[3ex] \frac{dS_{j,\mathrm{GABA}}}{dt} &= -\frac{S_{j,\mathrm{GABA}}}{\tau_\mathrm{GABA}} + \sum_{k \in
```

Comment:
The LaTeX does not render fully, see https://nest-simulator--2979.org.readthedocs.build/en/2979/models/iaf_wang_2002_exact.html